frontend/src/app/(protected)/experiments/add/components/AddExperimentContext.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import type React from "react";
+import { createContext, useContext, useState } from "react";
+import { ExperimentState } from "../../types";
+
+type ExperimentContextType = {
+  experimentState: ExperimentState;
+  setExperimentState: React.Dispatch<React.SetStateAction<ExperimentState>>;
+};
+
+const ExperimentContext = createContext<ExperimentContextType | undefined>(
+  undefined,
+);
+
+export const useExperiment = () => {
+  const context = useContext(ExperimentContext);
+  if (!context) {
+    throw new Error("useExperiment must be used within an ExperimentProvider");
+  }
+  return context;
+};
+
+export const ExperimentProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [experimentState, setExperimentState] = useState<ExperimentState>({
```

Comment:
Keeps track of the details for the experiment being created. It is available as a context so you don't need to pass it to subcomponents.